client/client.go

```diff
@@ -538,3 +542,32 @@ func (c *Client) Collect(ch chan<- prometheus.Metric) {
 var (
 	_ prometheus.Collector = (*Client)(nil)
 )
+
+// argListFromMySqlParams creates an array of strings from the pointer to the parameters for pt-mysql-sumamry
+func argListFromMySqlParams(pParams *agentpb.StartActionRequest_PTMySQLSummaryParams) []string {
+	var args []string
+
+	// Only adds the arguments are valid
+	// If socket valid, socket will be used. Otherwise it will try user and password.
+	if pParams.Socket != "" {
+		args = append(args, "--socket="+pParams.Socket)
+	} else {
+		if pParams.Username != "" {
+			args = append(args, "--user="+pParams.Username)
+		}
+
+		if pParams.Password != "" {
+			args = append(args, "--password="+pParams.Password)
```

Comment:
Should we escape passwords? What if the password has a `"`

Reply:
Can you specify what exactly you would like to do there?

Reply:
You using " and Carlos mention that password can also contain ". That is problem.

Reply:
I'm not sure if I understand it well here. "" stands for an empty string. If the password contains ", it will be handled as a string. It may generate an error when passed to pt-mysql-summary. If it does, it will finish with an error.
My question is: What do you desire to do here? Replace, remove and/or generate an error code without execution?

Reply:
A MySQL password can contain quotes and special chars. If we use just a `+` to concatenate the password, it might not work. 
Here is an example: https://play.golang.org/p/qlPjVEltESM

Reply:
Carlos, will pt-mysql-summary accept `--password mycoolpass`, without `=`? If yes, we should append two strings to args: `args = append(args, "--password", pParams.Password)`. os/exec.Cmd will handle the rest

Reply:
:+1: Yes, it would. @yareach  could you make it work as @AlekSi said?
Thanks.

Reply:
Roger that.